pandas/core/indexes/base.py

```diff
@@ -1993,7 +1993,11 @@ def __nonzero__(self):
     def __contains__(self, key):
         hash(key)
         try:
-            return key in self._engine
+            if (is_float(key) and is_integer_dtype(self.dtype) and
+                    int(key) != key):
+                return False
+            else:
+                return key in self._engine
```

Comment:
this is ok in python code, but actually can you see if you can do this in the cython code?

Reply:
Thanks. But I'm not good at Cython now. I'd like to try it but it'll take time.